lib/Zonemaster/Backend/TestAgent.pm

```diff
@@ -158,6 +101,34 @@ sub run {
         Zonemaster::Engine::Profile->effective->set( q{net.ipv6}, ( $params->{ipv6} ) ? ( 1 ) : ( 0 ) );
     }
 
+    if ( $show_progress ) {
+        my %methods = Zonemaster::Engine->all_methods;
+
+        # BASIC methods are always run: Basic0{0..4}
+        my $nbr_testcases_planned = 5;
+        my $nbr_testcases_finished = 0;
+
+        foreach my $module ( keys %methods ) {
+            foreach my $method ( @{ $methods{$module} } ) {
+                if ( Zonemaster::Engine::Util::should_run_test( $method ) ) {
+                    $nbr_testcases_planned++;
+                }
+            }
+        }
+
+        Zonemaster::Engine->logger->callback(
+            sub {
+                my ( $entry ) = @_;
+                if ( $entry->{tag} and $entry->{tag} eq 'TEST_CASE_END' ) {
+                    $nbr_testcases_finished++;
```

Comment:
Maybe check here that $entry->{testcase} is actually included in the profile? Otherwise we risk overshooting our 100%, like @matsduf says.